lib/list/getCSVData.js

```diff
@@ -0,0 +1,110 @@
+var _ = require('lodash');
+var listToArray = require('list-to-array');
+
+/**
+ * Applies option field transforms to get the CSV value for a field
+ */
+
+function transformFieldValue (field, item, options) {
+	var transform = typeof field.options.toCSV === 'string'
+		? listToArray(field.options.toCSV)
+		: field.options.toCSV;
+	if (typeof transform === 'function') {
+		return transform.call(item, field, options);
+	}
+	if (Array.isArray(transform)) {
+		var value = item.get(field.path);
+		if (transform.length === 1) {
+			return value[transform[0]];
+		} else {
+			return _.pick(value, transform);
+		}
+	}
+	return field.format(item);
+}
+
+/**
+ * Gets the data from an Item ready to be serialised to CSV for download
+ */
+
+function getCSVData (item, options) {
+	if (!options) {
+		options = {};
+	}
+	options.fields;
+	if (options.fields === undefined) {
+		options.fields = Object.keys(this.options.fields);
```

Comment:
Why would `options.fields` be undefined but `this.options.fields` defined? Where do those values come from?